### DISCUSSION TITLE
.env variables not working in development

### DISCUSSION THREAD

**Discussion by obolland** [vercel/next.js#21591] [Help]
**What version of Next.js are you using?**

10.0.5

**What version of Node.js are you using?**

15.4.0

**What browser are you using?**

Chrome

**What operating system are you using?**

macOS

**How are you deploying your application?**

Vercel

**Describe the Bug**

I have a local project and I cannot get .env files to load.

.env.local
`NEXT_PUBLIC_CODE=1234`

example.js
```
const Example = () => { return <h1>My code is {process.env.NEXT_PUBLIC_CODE}</h1> }
export default Example;
```

Have also tried loading them into a node environment, with and without the 'NEXT_PUBLIC_' prefix and they are undefined.

Any ideas?

**Expected Behavior**

I expected the .env variables to load.

**To Reproduce**

.env.local
`NEXT_PUBLIC_CODE=1234`

example.js
```
const Example = () => { return <h1>My code is {process.env.NEXT_PUBLIC_CODE}</h1> }
export default Example;
```

I have a more complicated project then the example above, and noticed the issue when I began integrating authentication.
However I have simplified it and run only the example above and the variable remains undefined.
If I install dotenv via npm then I can create a .env file and load the variable successfully, however this introduces other errors.

**Comment by samullman** [comment]
Yup having the same issue ... It _wasn't_ working, then it _was_, now it's not 😬

**Next** 10.0.4
**Node** v12.18.2
**Browser** Brave
**OS** macOS

**Reply by samullman** [reply]
Fixed !!! @obolland 
```
const client = require('contentful').createClient({
  space: process.env.CONTENTFUL_SPACE_ID ,
  accessToken: process.env.CONTENTFUL_ACCESS_TOKEN,
});
```
to
```
const client = require('contentful').createClient({
  space: `${process.env.CONTENTFUL_SPACE_ID}` ,
  accessToken: `${process.env.CONTENTFUL_ACCESS_TOKEN}`,
});
```

**Reply by samullman** [reply]
Negative, broken in Prod ... Welp

And Dev ... No idea 🤷

**Reply by samullman** [reply]
console.log( process.env.CONTENTFUL_SPACE_ID )
// 👍👍👍👍

**Comment by gargpuja** [comment]
You can try creating next.config.js file 
and copy paste
```
module.exports = {
  env: {
    NEXT_PUBLIC_CODE: process.env.NEXT_PUBLIC_CODE,
  },
}
```

This may work
